addons/GeometricConstruct/src/presenter.js

```diff
@@ -0,0 +1,600 @@
+function AddonGeometricConstruct_create() {
+    var presenter = function () {};
+
+    presenter.labelsList = [];
+    presenter.figuresList = [];
+    presenter.toolbarButtonsDict = {};
+    presenter.newFigure = null;
+    presenter.editFigure = null;
+    presenter.isEditMode = false;
+    presenter.isDragging = false;
+
+    presenter.playerController = null;
+
+    presenter.CSS_CLASSES = {
+        TOOLBAR_WRAPPER: 'toolbar_wrapper',
+        WORKSPACE_WRAPPER: 'workspace_wrapper',
+        CANVAS_OVERLAY: 'canvas_overlay',
+        REMOVE_FIGURE_BUTTON: 'remove_figure',
+        CURSOR: 'Cursor',
+        CURSOR_IMAGE: 'cursor-image'
+    };
+
+    presenter.createPreview = function (view, model) {
+        presenterLogic(view, model, true);
+    };
+
+    presenter.run = function (view, model) {
+        presenterLogic(view, model, false);
+    };
+
+    function presenterLogic(view, model, isPreview) {
+        presenter.configuration = presenter.validateModel(model);
+        presenter.setElements(view);
+        presenter.createView(isPreview);
+        if (!presenter.configuration.defaultVisibility) presenter.hide();
+    }
+
+    presenter.show = function() {
+        presenter.$view.css('visibility', '');
+        presenter.$view.css('display', '');
+    }
+
+    presenter.hide = function() {
+        presenter.$view.css('visibility', 'hidden');
+        presenter.$view.css('display', 'none');
+    }
+
+    presenter.isVisible = function() {
+        return presenter.$view.css('display') != 'none' &&  presenter.$view.css('visibility') != 'hidden';
+    }
+
+    presenter.validateModel = function (model) {
+        console.log(model);
+        var strokeColor = (model["strokeColor"] && model["strokeColor"].trim().length > 0) ? model["strokeColor"] : "black";
+        var fillColor = (model["fillColor"] && model["fillColor"].trim().length > 0) ? model["fillColor"] : "blue";
```

Comment:
Myślałem trochę o tych kolorach i proponował bym ustalenie szczegółów z LEA czy ma być walidacja dla kolorów. W dokumentacji jest mowa o kolorze ale nie jest opisane jakie format może przyjąć. Przykład z addonu Animation: "Watermark color in #RRGGBB format (default is #000000 - black)"
Przynajmniej opis w dokumentacji byłby pomocny.